src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsPluginConfigurable.kt

```diff
@@ -68,9 +68,23 @@ class SettingsPluginConfigurable : Configurable {
         settingsState.buildPath = settingsComponent!!.buildPath
         settingsState.buildCommand = settingsComponent!!.buildCommand
         settingsState.telemetryEnabled = settingsComponent!!.telemetryEnabled
+        if (checkEmptyTelemetryPath(settingsState)) return
```

Comment:
Wouldn't it make sense to put this check at the start of the method?

Reply:
I thought about it. IntellIj refactor did it that way, I figured it might be more readable for the developer if they see that there is a potential return statement in the main code body rather than in the helper function. 
It is not clean, but it is obvious that the following line will not be executed if the check passes.

Reply:
Alright (after some in-person discussion)